File: 백준/Bronze/10810. 공 넣기/공 넣기.js
function solution(input) {
    let result = new Array(input[0][0]).fill(0);
    for(let i = 1; i <= input[0][1]; i++) {
        result = result.map((el, index) => index >= input[i][0] - 1 && index <= input[i][1] - 1 ? input[i][2] : el );
    }
    console.log(result.join(' '));
}

const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,    
});

let input = [];
let list = [];

rl.on("line", function (line) {
   input.push(line)
}).on("close", function () {
    input.forEach((val) => {
       list.push(val.split(' ').map((el) => parseInt(el)));
    });
    solution(list);
    process.exit();
});

File: 백준/Bronze/10810. 공 넣기/공 넣기.py
"""
바구니: N
실행 횟수: M
시간 제한: 1초

범위: i ~ k
범위에 같은 번호의 공을 넣는다.
"""

# 입력: N, M / 바구니 배열 생성
N, M = map(int, input().split())
basket = [0 for _ in range(N)]

# M번의 FOR문
for _ in range(M):
    i, j, k = map(int, input().split())
    # FOR문 I - K까지 번호 변경
    for idx in range(i - 1, j):
        basket[idx] = k

# 출력
print(" ".join(map(str, basket)))